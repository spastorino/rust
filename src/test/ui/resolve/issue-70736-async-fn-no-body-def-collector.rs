// edition:2018

async fn free(); //~ ERROR without a body

struct A;
impl A {
    async fn inherent(); //~ ERROR without body
}

trait B {
    async fn associated();
}
impl B for A {
    //~^ ERROR not all trait items implemented, missing: `__Assoc` [E0046]
    async fn associated(); //~ ERROR without body
    //~^ ERROR incompatible type for trait
}

fn main() {}
