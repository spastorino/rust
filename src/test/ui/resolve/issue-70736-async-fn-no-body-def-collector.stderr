error: free function without a body
  --> $DIR/issue-70736-async-fn-no-body-def-collector.rs:3:1
   |
LL | async fn free();
   | ^^^^^^^^^^^^^^^-
   |                |
   |                help: provide a definition for the function: `{ <body> }`

error: associated function in `impl` without body
  --> $DIR/issue-70736-async-fn-no-body-def-collector.rs:7:5
   |
LL |     async fn inherent();
   |     ^^^^^^^^^^^^^^^^^^^-
   |                        |
   |                        help: provide a definition for the function: `{ <body> }`

error: associated function in `impl` without body
  --> $DIR/issue-70736-async-fn-no-body-def-collector.rs:15:5
   |
LL |     async fn associated();
   |     ^^^^^^^^^^^^^^^^^^^^^-
   |                          |
   |                          help: provide a definition for the function: `{ <body> }`

error[E0053]: method `associated` has an incompatible type for trait
  --> $DIR/issue-70736-async-fn-no-body-def-collector.rs:15:26
   |
LL |     async fn associated();
   |                          - type in trait
...
LL |     async fn associated();
   |                          ^
   |                          |
   |                          checked the `Output` of this `async fn`, found opaque type
   |                          expected `()`, found opaque type
   |
   = note: while checking the return type of the `async fn`
   = note: expected fn pointer `fn()`
              found fn pointer `fn() -> impl Future`

error[E0046]: not all trait items implemented, missing: `__Assoc`
  --> $DIR/issue-70736-async-fn-no-body-def-collector.rs:13:1
   |
LL |     async fn associated();
   |     ---------------------- `__Assoc` from trait
LL | }
LL | impl B for A {
   | ^^^^^^^^^^^^ missing `__Assoc` in implementation

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0046, E0053.
For more information about an error, try `rustc --explain E0046`.
